app/javascript/components/collections/AddToCollections.vue

```diff
@@ -0,0 +1,183 @@
+                {{ $t('collections.edit.addCollection') }}
+              </p>
+              <el-select
+                v-model="collectionsIdsInput"
+                :placeholder="this.$t('all.select')"
+                filterable
+                size="large"
+                class="w-full"
+              >
+                <el-option
+                  v-for="item in collectionsList"
+                  :key="item.id"
+                  :label="item.name"
+                  :value="item.id"
+                />
+              </el-select>
+            </div>
+          </div>
+        </div>
+      </div>
+      <template #footer>
+        <span class="flex justify-between gap-3 px-5">
+          <div
+            class="w-[50%] active:outline active:outline-4 active:outline-[#EAECF0] hover:bg-[#F0F2F5] text-center py-[10px] px-4 border border-[#D0D5DD] cursor-pointer shadow-xs rounded-[8px]"
+            @click="cancelAddCollections"
+          >
+            {{ $t('all.cancel') }}
+          </div>
+          <div
+            class="w-[50%] hover:bg-[#223B99] hover:border-[#223B99] active:outline active:outline-4 active:outline-[#D4DbF5] text-[#ffffff] text-center py-[10px] px-4 border border-[#3250BD] bg-[#3250BD] cursor-pointer shadow-xs rounded-[8px]"
+            @click="confirmAddCollections"
+          >
+            {{ $t('all.confirm') }}
+          </div>
+        </span>
+      </template>
+    </el-dialog>
+  </div>
+</template>
+<script setup>
+  import { ref, inject, onMounted } from 'vue'
+  import jwtFetch from '../../packs/jwtFetch'
+  import { ElMessage } from 'element-plus'
+  import { useI18n } from 'vue-i18n'
+
+  const { t } = useI18n()
+
+  const csghubServer = inject('csghubServer')
+
+  const props = defineProps({
+    repoId: String,
+    userName: String,
+    settingsVisibility: Boolean
+  })
+  const dialogVisible = ref(false)
+  const collectionsList = ref([])
+  const collectionsIdsInput = ref('')
+
+  const fetchCollectionsList = async () => {
+    const url = `${csghubServer}/api/v1/user/${props.userName}/collections`
+    const res = await jwtFetch(url)
+    if (!res.ok) {
+      const { msg } = await res.json()
+      ElMessage({ message: msg, type: 'warning' })
+    } else {
+      const { data } = await res.json()
+      collectionsList.value = data
+    }
+  }
+
+  const confirmAddCollections = () => {
+    if (!collectionsIdsInput.value) {
+      ElMessage({ message: t('all.selectProject'), type: 'warning' })
+      return
+    }
+    repoAddToCollections()
+  }
+
+  const cancelAddCollections = () => {
+    dialogVisible.value = false
+    collectionsIdsInput.value = ''
+  }
+
+  async function repoAddToCollections() {
+    const addRepoData = {
+      repo_ids: [props.repoId]
+    }
+    const options = { method: 'POST', body: JSON.stringify(addRepoData) }
+    const url = `${csghubServer}/api/v1/collections/${collectionsIdsInput.value}/repos`
+    const response = await jwtFetch(url, options)
```

Comment:
please pull latest main branch code and update jwtFetch to the new method

Reply:
down